### PR TITLE
CORE-9012 Use custom Avro type for response to FindUnconsumedStatesByType

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Added avro messages for ledger transaction outputs

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/TransactionOutput.avsc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "type": "record",
+  "name": "TransactionOutput",
+  "doc": "Transaction output data of signed transaction.",
+  "namespace": "net.corda.data.ledger.persistence",
+  "fields": [
+    {
+        "name": "id",
+        "type": "string",
```
Comment (vlajos) [inline data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/TransactionOutput.avsc]: Would not be net.corda.data.crypto.SecureHash better for this?

Comment (mbrkic-r3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/TransactionOutput.avsc]: I've kept it consistent with types we used so far (e.g. as in https://github.com/corda/corda-api/blob/release/os/5.0/data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/FindTransaction.avsc)

**Comment by vlajos** [comment]
I think we should add utxo somewhere to the name/package.
These are quite UTXO specific.

**Comment by mbrkic-r3** [comment]
> I think we should add utxo somewhere to the name/package. These are quite UTXO specific.

renamed

**Review by vlajos — APPROVED** [review]

**Pull request merged**